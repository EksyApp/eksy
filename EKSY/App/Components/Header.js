import React, {Component} from 'react'
import * as Theme from '../Theme'
import {StyleSheet, View, Text} from 'react-native'
import MenuButton from './MenuButton'

console.log(Theme)

class Header extends Component {

  render() {
    return(
      <View style={styles.container}>
        <MenuButton />
        <Text style={styles.text}>{this.props.title}</Text>
      </View>
    )
  }

}

const styles = StyleSheet.create({

  container: {
    height: 50,
    backgroundColor: Theme.FrontgroundColor,
    justifyContent: 'center'
  },

  text: {
    alignSelf: 'center',
    fontSize: 18,
    color: Theme.FontColor
  }

})

export default Header
