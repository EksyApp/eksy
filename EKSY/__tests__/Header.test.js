import 'react-native';
import React from 'react';
import Header from '../App/Components/Header';
import renderer from 'react-test-renderer';

test('header renders correctly with backbutton value true', () => {
  const tree = renderer.create(
    <Header backButton/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('header renders correctly with backbutton value false', () => {
  const tree = renderer.create(
    <Header />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
