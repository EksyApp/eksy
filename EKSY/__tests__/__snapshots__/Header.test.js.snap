// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly with backbutton value false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(226,227,217)",
      "height": 50,
      "justifyContent": "center",
    }
  }
>
  <View
    onPress={undefined}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        Object {
          "height": 50,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 50,
          "zIndex": 1,
        },
      ]
    }
    underlayColor="white"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 35,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "rgb(101,114,122)",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`header renders correctly with backbutton value true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(226,227,217)",
      "height": 50,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          Object {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 50,
            "zIndex": 1,
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 35,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "rgb(101,114,122)",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  />
</View>
`;
